org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/rule/RuleDetailsPanel.java

```diff
@@ -109,40 +99,52 @@ private void updateScrollCompositeMinSize() {
+  
+  public void updateRule(AbstractRuleDto ruleInformation, Either<RuleMonolithicDescriptionDto, RuleSplitDescriptionDto> description) {
     try {
-      var ruleDefinition = getStandaloneRuleDescriptionResponse.getRuleDefinition();
-
-      ruleNameLabel.setText(ruleDefinition.getName());
+      ruleNameLabel.setText(ruleInformation.getName());
       ruleNameLabel.requestLayout();
-      ruleHeaderPanel.updateRule(ruleDefinition.getKey(), ruleDefinition.getType(), ruleDefinition.getDefaultSeverity());
 
-      updateHtmlDescription(getStandaloneRuleDescriptionResponse.getDescription(), ruleDefinition.getLanguage().getLanguageKey());
+      updateHeader(ruleInformation);
+
+      updateHtmlDescription(description, ruleInformation.getLanguage().getLanguageKey());
+      
+      if (ruleInformation instanceof EffectiveRuleDetailsDto) {
+        updateParameters((EffectiveRuleDetailsDto) ruleInformation);
+      }
 
       requestLayout();
       updateScrollCompositeMinSize();
     } catch (SWTException ignored) {
       // There might be a race condition between the background job running late and the view already being closed
     }
   }
-
-  public void updateRule(GetEffectiveRuleDetailsResponse getEffectiveRuleDetailsResponse) {
-    try {
-      var details = getEffectiveRuleDetailsResponse.details();
-
-      ruleNameLabel.setText(details.getName());
-      ruleNameLabel.requestLayout();
-      ruleHeaderPanel.updateRule(details.getKey(), details.getType(), details.getSeverity());
-
-      updateHtmlDescription(details.getDescription(), details.getLanguage().getLanguageKey());
-      updateParameters(details);
-
-      requestLayout();
-      updateScrollCompositeMinSize();
-    } catch (SWTException ignored) {
-      // There might be a race condition between the background job running late and the view already being closed
+  
+  private void updateHtmlDescription(Either<RuleMonolithicDescriptionDto, RuleSplitDescriptionDto> description, String languageKey) {
+    if (ruleDescriptionPanel != null && !ruleDescriptionPanel.isDisposed()) {
+      ruleDescriptionPanel.dispose();
     }
+    ruleDescriptionPanel = new RuleDescriptionPanel(scrolledContent, languageKey, useEditorFontSize);
+    ruleDescriptionPanel.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 1, 1));
+    ruleDescriptionPanel.updateRule(description);
+  }
+  
+  private void updateHeader(AbstractRuleDto ruleInformation) {
+    if (ruleHeaderPanel != null && !ruleDescriptionPanel.isDisposed()) {
+      ruleHeaderPanel.dispose();
+    }
+    
+    var attributeOptional = ruleInformation.getCleanCodeAttribute();
+    var impacts = ruleInformation.getDefaultImpacts();
+    
+    if (attributeOptional.isPresent() && !impacts.isEmpty()) {
+      ruleHeaderPanel = new RuleHeaderPanel(scrolledContent);
+    } else {
+      ruleHeaderPanel = new LegacyRuleHeaderPanel(scrolledContent);
```

Comment:
I'm not familiar with Eclipse code but if this method is purely for the rule panel (not the issue panel), then we only want to show the new CCT. In any case, we will always have the new analyzer in the upcoming release, so we should expect only new CCT for rules.

The only case with old CCT is with SQ <10.2, which means it's only for issues in connected mode

Reply:
No, everything in Eclipse is shared 😄 This is reused in the **Rule Description view** (for the issues) and the **Rules Configuration settings panel** (for the rules).